CMakeLists.txt

```diff
@@ -0,0 +1,141 @@
+else()
+    set(LIBXMP_PIC OFF)
+endif()
+# -----------------------------------
+
+if(EMSCRIPTEN OR VITA)
+    set(BUILD_SHARED_DEFAULT OFF)
+else()
+    set(BUILD_SHARED_DEFAULT ON)
+endif()
+
+option(BUILD_STATIC   "Build static library of libXMP" ON)
+option(BUILD_SHARED   "Build shared library of libXMP" ${BUILD_SHARED_DEFAULT})
+if(NOT BUILD_STATIC AND NOT BUILD_SHARED)
+    message(FATAL_ERROR "Both static and shared builds got disabled. You must enable at least one of them.")
+endif()
+
+option(LIBXMP_DISABLE_DEPACKERS     "Disable archive depackers at libXMP" OFF)
+option(LIBXMP_DISABLE_PROWIZARD     "Disable the ProWizard at libXMP" OFF)
+
+include(build_src.cmake)
+
+if(LIBXMP_DISABLE_DEPACKERS)
+    list(APPEND LIBXMP_DEFINES_PUBLIC -DLIBXMP_NO_DEPACKERS)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL ${LIBXMP_SRC_LIST_DEPACKERS})
+endif()
+
+if(LIBXMP_DISABLE_PROWIZARD)
+    list(APPEND LIBXMP_DEFINES_PUBLIC -DLIBXMP_NO_PROWIZARD)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL ${LIBXMP_SRC_LIST_PROWIZARD})
+endif()
+
+if(MSVC)
+    list(APPEND LIBXMP_DEFINES -Dinline=__inline -D_USE_MATH_DEFINES -DPATH_MAX=260)
+endif()
+
+
+list(APPEND LIBXMP_SRC_FULL ${LIBXMP_SRC_LIST_LITE} ${LIBXMP_SRC_LIST_FULL} ${LIBXMP_SRC_COMMON})
+
+set(XMP_INSTALLS)
+
+# ============= FULL =============
+if(BUILD_STATIC)
+    add_library(xmp_static STATIC ${LIBXMP_SRC_FULL})
+    list(APPEND XMP_INSTALLS xmp_static)
+    set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp C_STANDARD 90)
+
+    if(MSVC AND BUILD_SHARED)
+        set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp-static)
+    else()
+        set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp)
+    endif()
+
+    target_compile_options(xmp_static PRIVATE ${LIBXMP_CFLAGS})
+    target_compile_definitions(xmp_static PRIVATE ${LIBXMP_DEFINES} -DBUILDING_STATIC)
+    target_compile_definitions(xmp_static PUBLIC ${LIBXMP_DEFINES_PUBLIC})
+
+    if(LIBXMP_PIC)
+        target_compile_options(xmp_static PRIVATE -fPIC)
+    endif()
+
+    target_include_directories(xmp_static PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+    target_include_directories(xmp_static PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
+
+    if(LIBM_REQUIRED)
+        target_link_libraries(xmp_static PUBLIC ${LIBM_LIBRARY})
+    endif()
+endif()
+
+if(BUILD_SHARED)
+    add_library(xmp_shared SHARED ${LIBXMP_SRC_FULL})
+    list(APPEND XMP_INSTALLS xmp_shared)
+    set_target_properties(xmp_shared PROPERTIES OUTPUT_NAME xmp C_STANDARD 90)
+
+    if(NOT MSVC)
+        target_link_libraries(xmp_shared PUBLIC m)
+    endif()
+
+    if(HAVE_VISIBILITY AND UNIX AND (NOT APPLE))
+        set_property(TARGET xmp_shared APPEND_STRING PROPERTY
+                     LINK_FLAGS "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/libxmp.map")
+    endif()
+
+    target_compile_options(xmp_shared PRIVATE ${LIBXMP_CFLAGS})
+    target_compile_definitions(xmp_shared PRIVATE ${LIBXMP_DEFINES} -DBUILDING_DLL)
```

Comment:
This should be `WIN32` only -- harmless, but looks funny in linux or mac builds

Reply:
No problems